test/ffmpeg-qsv/transcode/transcoder.py

```diff
@@ -75,11 +84,19 @@ def validate_spec(self):
     for output in self.outputs:
       codec = output["codec"]
       mode  = output["mode"]
+      width = output.get("width", 0)
+      height  = output.get("height", 0)
       assert mode in ["sw", "hw"], "Invalid test case specification as output mode type not valid"
       oplats, oreq, _ = self.get_requirements_data("encode", codec, mode)
       platforms &= set(oplats)
       requires.append(oreq)
 
+      if width > 0 and height > 0:
```

Comment:
What if user only specify width and not height (or vice versa)?  Furthermore, if such a case, either one should default to source dimension anywhere it is used in the generated transcode pipeline commands.

Reply:
instead of falling back to source dimensions, shall I assert as width or height invalid value?